scipy/optimize/tests/test_quadratic_assignment.py

```diff
@@ -69,13 +68,19 @@ def test_accuracy_1(self):
              [0, 2, 0, 2],
              [0, 1, 2, 0]]
 
-        res = quadratic_assignment(A, B, method=self.method,
-                                   options={"rng": 0, "maximize": False})
+        with assert_warns(FutureWarning):
+            res = quadratic_assignment(A, B, method=self.method,
+                                       options={"rng": 0, "maximize": False})
```

Comment:
Can we `match` the message and add a test for the other FutureWarning and the DeprecationWarning?